app/src/main/java/br/com/alexf/boraprofut/features/drawTeams/useCases/TeamDrawerUseCase.kt

```diff
@@ -1,48 +1,79 @@
 package br.com.alexf.boraprofut.features.drawTeams.useCases
 
 import br.com.alexf.boraprofut.models.Player
-import kotlin.math.ceil
 
 class TeamDrawerUseCase {
 
     fun drawRandomTeams(
         players: Set<Player>,
         playersPerTeam: Int
     ): List<Set<Player>> {
-        return players
-            .shuffled()
-            .chunked(playersPerTeam) {
-                it.toSet()
-            }
+        return getTeams(players = players, playersPerTeam = playersPerTeam)
     }
 
     fun drawBalancedTeams(
         players: Set<Player>,
         playersPerTeam: Int
     ): List<Set<Player>> {
-        val amountTeams = ceil(players.size / playersPerTeam.toFloat()).toInt()
-        val sortedPlayers = players.shuffled()
-            .sortedBy {
-                it.level
-            }.toMutableList()
-
-        val teamsWithBestPlayersIncluded = MutableList(amountTeams) {
-            val bestPlayer = sortedPlayers.removeLast()
-            mutableListOf(bestPlayer)
+        return getTeams(isBalancedTeam = true, players = players, playersPerTeam = playersPerTeam)
+    }
+
+    private fun getTeams(
+        isBalancedTeam: Boolean = false,
+        players: Set<Player>,
+        playersPerTeam: Int
+    ): List<Set<Player>> {
+        val onlyGoalKeepers = players
+            .filter { it.isGoalKeeper }
+            .shuffled()
+            .toMutableList()
+        val onlyPlayers = if (isBalancedTeam) {
+            players.filter { !it.isGoalKeeper }.sortedByDescending { it.level }.toMutableList()
+        } else {
+            players.filter { !it.isGoalKeeper }.shuffled().toMutableList()
         }
+        val totalOfGoalKeepers = onlyGoalKeepers.size
+        val totalOfTeams: Int = players.size / playersPerTeam
+        val teams: MutableList<MutableList<Player>> =
+            MutableList(totalOfTeams) { mutableListOf() }
+        var firstTeamWithoutGoalKeeper = 0
 
-        while (sortedPlayers.isNotEmpty()) {
-            teamsWithBestPlayersIncluded.forEach {
-                if (sortedPlayers.isEmpty()) {
-                    return@forEach
+        // add players to the teams
+        if (teams.isNotEmpty()) {
+            for (current in 0 until playersPerTeam - 1) {
```

Comment:
para não ter que lidar com isso, to pensando em colocar deixar a compreensão da quantidade de jogadores apenas para jogadores e os goleiros entra como um jogador que não conta... O que acha? Pq vai ficar estranho os times com '5 jogadores' sendo que só tem 4 kkkkkk

Reply:
Irei fazer essa modificação, tinha considerado que no momento de definição da quantidade de jogadores em um time, estivesse contando com o goleiro.